controller-server/src/main/java/com/yahoo/vespa/hosted/controller/tenant/CloudTenant.java

```diff
@@ -36,6 +39,8 @@ public CloudTenant(TenantName name, Instant createdAt, LastLoginInfo lastLoginIn
         this.info = Objects.requireNonNull(info);
         this.tenantSecretStores = tenantSecretStores;
         this.archiveAccessRole = archiveAccessRole;
+        if (!archiveAccessRole.map(role -> VALID_ARCHIVE_ACCESS_ROLE_PATTERN.matcher(role).matches()).orElse(true))
+            throw new IllegalArgumentException("Invalid archive access role name: " + archiveAccessRole.get());
```

Comment:
Great with additional validation - maybe add the pattern string to the error message so it's easier to find the problem?